src/filters/simple-filter/filter-component.js

```diff
@@ -33,18 +32,28 @@ angular.module('patternfly.filters').component('pfFilter', {
         type: field.filterType,
         value: value
       };
-      if (!filterExists(newFilter)) {
+      var matchedFilter;
+
+      if (filterExists(newFilter)) {
 
         if (newFilter.type === 'select') {
           enforceSingleSelect(newFilter);
         }
 
+        if (field.filterType === 'complex-select' && !field.filterMultiselect && _.some(ctrl.config.appliedFilters, complexFilterExists)) {
+          _.remove(ctrl.config.appliedFilters, matchedFilter);
+        }
+
         ctrl.config.appliedFilters.push(newFilter);
 
         if (ctrl.config.onFilterChange) {
           ctrl.config.onFilterChange(ctrl.config.appliedFilters);
         }
       }
+
+      function complexFilterExists (item) {
+        return _.includes(item.value, _.split(value, field.filterDelimiter, 1)) ? matchedFilter = item : null;
+      }
```

Comment:
The logic of this in kinda janky.  A function called 'complexFilterExists' should return true/false without a side effect of setting the matchedFilter variable which is then used.